utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -21,7 +24,24 @@ type Client struct {
 	pk     *bls.PublicKey
 }
 
-func NewClient(ctx context.Context, conn *grpc.ClientConn) (*Client, error) {
+func NewClient(ctx context.Context, url string) (*Client, error) {
+	// TODO: figure out the best parameters here given the target block-time
+	opts := grpc.WithConnectParams(grpc.ConnectParams{
+		Backoff: backoff.DefaultConfig,
+	})
+
+	// the rpc-signer client should call a proxy server (on the same machine) that forwards
+	// the request to the actual signer instead of relying on tls-credentials
+	conn, err := grpc.NewClient(url, opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rpc signer client: %w", err)
+	}
+	defer func() {
+		if err != nil {
```

Comment:
The typical way to clarify this is to change the function signature to:

```go
func NewClient(ctx context.Context, url string) (_ *Client, retErr error) {
```

and then here

```go
defer func() {
    if retErr != nil {
        conn.Close()
    }
}()
```

You'll never explicitly set `retErr` so it can't be accidentally overwritten and is known to be the returned error.